scripts/Muon/GUI/Common/load_utils.py

```diff
@@ -84,10 +101,387 @@ def getWorkspaceNames(self):
     # Get the groups/pairs for active WS
     def getGroupedWorkspaceNames(self):
         # gets all WS in the ADS
-        runName,options = self.getCurrentWS()
-        final_options=[]
+        runName, options = self.getCurrentWS()
+        final_options = []
         # only keep the relevant WS (same run as Muon Analysis)
         for pick in options:
             if "MuonAnalysisGrouped_" in pick and ";" not in pick:
                 final_options.append(pick)
         return final_options
+
+
+def get_default_instrument():
+    default_instrument = ConfigServiceImpl.Instance().getInstrument().name()
+    if default_instrument not in file_utils.allowed_instruments:
+        default_instrument = "None"
+    return default_instrument
+
+
+def run_LoadInstrument(parameter_dict):
+    """
+    Apply the LoadInstrument algorithm with the properties supplied through
+    the input dictionary of {proeprty_name:property_value} pairs.
+    Returns the calculated value of alpha.
+    """
+    alg = mantid.AlgorithmManager.create("LoadInstrument")
+    alg.initialize()
+    alg.setAlwaysStoreInADS(False)
+    # alg.setProperty("Workspace", "__notUsed")
+    alg.setProperties(parameter_dict)
+    alg.execute()
+    return alg.getProperty("Workspace").value
+
+
+def __default_workspace():
+    default_instrument = get_default_instrument()
+    workspace = api.WorkspaceFactoryImpl.Instance().create("Workspace2D", 2, 10, 10)
+    workspace = run_LoadInstrument(
+        {"Workspace": workspace,
+         "RewriteSpectraMap": True,
+         "InstrumentName": default_instrument})
+    return MuonWorkspace(workspace)
+
+
+# Dictionary of (property name):(property value) pairs to put into Load algorithm
+# NOT including "OutputWorkspace" and "Filename"
+DEFAULT_INPUTS = {
+    "DeadTimeTable": "__notUsed",
+    "DetectorGroupingTable": "__notUsed"}
+# List of property names to be extracted from the result of the Load algorithm
+DEFAULT_OUTPUTS = ["OutputWorkspace",
+                   "DeadTimeTable",
+                   "DetectorGroupingTable",
+                   "TimeZero",
+                   "FirstGoodData",
+                   "MainFieldDirection"]
+# List of default values for the DEFAULT_OUTPUTS list
+DEFAULT_OUTPUT_VALUES = [__default_workspace(),
+                         None,  # api.WorkspaceFactoryImpl.Instance().createTable("TableWorkspace"),
+                         api.WorkspaceFactoryImpl.Instance().createTable("TableWorkspace"),
+                         0.0,
+                         0.0, "Unknown direction"]
+
+
+class floatPropertyWithValue(object):
```

Comment:
do you know where this is used? I dont understand its purpose from the description

Reply:
This is I think used when initializing with default values in MuonLoader. I think it is used to make a dummy workspace look like a workspace.

Reply:
could we give it a better description? I think the default values should come from the context.

Reply:
Having double checked I can't find anywhere where it is currently used. I thought I had found a usage but it doesn't seem to be there. I will remove for now.